File: src/api/Models/MinecraftEnchantment.cs
namespace MinecraftDataApi.Models;

public class MinecraftEnchantment
{
    public string Name { get; set; } = string.Empty;
    public int MaxLevel { get; set; }
    public string Description { get; set; } = string.Empty;
    public List<string> ApplicableItems { get; set; } = new List<string>();
    public bool LibrariansCanTrade { get; set; }
}

public class MinecraftEnchantmentState
{
    public bool HasLibrarianTrade { get; set; }
    public int? Level { get; set; }
    public int? EmeraldCost { get; set; }
}

File: minecraft-tracker-ui/src/types/MinecraftEnchantment.ts
// Static enchantment data
export interface MinecraftEnchantment {
  name: string;
  maxLevel: number;
  description: string;
  applicableItems: string[];
  librariansCanTrade: boolean;
}

// Mutable enchantment state
export interface MinecraftEnchantmentState {
  hasLibrarianTrade: boolean;
  level?: number;
  emeraldCost?: number;
}

// Combined type for UI display
export interface EnchantmentViewModel extends MinecraftEnchantment {
  hasLibrarianTrade: boolean;
  level?: number;
  emeraldCost?: number;
}